File: samples/golang/internal/client/credentials.go
// Package client 提供 VibeOps Agent 客户端的公共实现
package client

import (
	"fmt"

	credentials "github.com/aliyun/credentials-go/credentials"
)

// LoadCredentialsFromChain 通过阿里云默认凭据链获取凭证
// 凭据链优先级：环境变量 > OIDC > CLI配置文件 > 配置文件(~/.alibabacloud/credentials) > IAM角色
func LoadCredentialsFromChain() (accessKeyID, accessKeySecret string, err error) {
	cred, err := credentials.NewCredential(nil)
	if err != nil {
		return "", "", fmt.Errorf("初始化凭据失败: %w", err)
	}

	credValue, err := cred.GetCredential()
	if err != nil {
		return "", "", fmt.Errorf("获取凭证失败: %w", err)
	}

	if credValue.AccessKeyId == nil || credValue.AccessKeySecret == nil {
		return "", "", fmt.Errorf("凭据链返回的凭证为空")
	}

	return *credValue.AccessKeyId, *credValue.AccessKeySecret, nil
}

File: samples/typescript/src/client/credentials.ts
/**
 * 默认凭据链支持 - 与 Go 实现保持一致
 * 凭据链优先级：环境变量 > OIDC > CLI配置文件 > 配置文件(~/.alibabacloud/credentials) > IAM角色
 */
import CredentialPkg from '@alicloud/credentials';

type CredentialValue = {
  accessKeyId?: string;
  accessKeySecret?: string;
};

type CredentialClient = {
  getCredential(): Promise<CredentialValue>;
};

type CredentialConstructor = new () => CredentialClient;

function resolveCredentialConstructor(pkg: unknown): CredentialConstructor {
  const moduleValue = pkg as Record<string, unknown> | undefined;
  const defaultValue = moduleValue?.default as Record<string, unknown> | undefined;
  const candidates = [pkg, moduleValue?.default, defaultValue?.default];

  for (const candidate of candidates) {
    if (typeof candidate === 'function') {
      return candidate as CredentialConstructor;
    }
  }

  throw new TypeError('@alicloud/credentials 未导出可用的 Credential 构造函数');
}

export async function loadCredentialsFromChain(): Promise<{ accessKeyId: string; accessKeySecret: string }> {
  const Credential = resolveCredentialConstructor(CredentialPkg);
  const cred = new Credential();
  const credential = await cred.getCredential();
  const accessKeyId = credential.accessKeyId;
  const accessKeySecret = credential.accessKeySecret;
  if (!accessKeyId || !accessKeySecret) {
    throw new Error('凭据链返回的凭证为空');
  }
  return { accessKeyId, accessKeySecret };
}
